generators/chipyard/src/main/scala/IOBinders.scala

```diff
@@ -262,16 +271,16 @@ class WithSimNIC extends OverrideIOBinder({
 // accessible to the IOBinder
 // DOC include start: WithSimAXIMem
 class WithSimAXIMem extends OverrideIOBinder({
-  (system: CanHaveMasterAXI4MemPort with BaseSubsystem) => {
-    val peiTuples = AddIOCells.axi4(system.mem_axi4, system.memAXI4Node)
+  (system: CanHaveMasterAXI4MemPort, p) => {
+    val peiTuples = AddIOCells.axi4(system.mem_axi4, system.memAXI4Node, "mem")
     // TODO: we are inlining the connectMem method of SimAXIMem because
     //   it takes in a dut rather than seq of axi4 ports
     val harnessFn = (th: chipyard.TestHarness) => {
       peiTuples.map { case (port, edge, ios) =>
-        val mem = LazyModule(new SimAXIMem(edge, size = system.p(ExtMem).get.master.size)(system.p))
+        val mem = LazyModule(new SimAXIMem(edge, size = p(ExtMem).get.master.size)(p))
```

Comment:
One other thought- would some kind of match work?

```suggestion
        val p = system match {
          case BaseSubsystem => system.p
          case _ => assert(false)
        }
        val mem = LazyModule(new SimAXIMem(edge, size = p(ExtMem).get.master.size)(p))
```

Reply:
The new version does this basically.

Reply:
I was still thinking you'd call the object function here, rather than pass `p` as an argument

Reply:
Ok, I can do it that way.